subcommands/git/cmd.go

```diff
@@ -73,8 +74,16 @@ func doGitCreds(cmd *cobra.Command, args []string) {
 	helperName := "fio"
 	dst := filepath.Join(helperPath, GIT_CREDS_HELPER)
 	if runtime.GOOS == "windows" {
-		dst += ".exe"
-		helperName += ".exe"
+		/*
+		** On Windows should use the absolute path.
+		** If you install git for windows, there is an option to not support symlinks everywhere.
+		** For some reason it will not run the configured credential helper command correctly.
+		** When using an absolute path the `\` should be replaced for `/` characters.
+		** In addition to that, spaces ` ` should be encoded to `\\ `.
+		** So for example the following path will be used: C:/Program\\ Files/Git/bin/git-credential-fio.exe
+		 */
+		dst = strings.ReplaceAll(strings.ReplaceAll(dst, "\\", "/"), " ", "\\\\ ") + ".exe"
+		helperName = dst
```

Comment:
hmm... this is interesting...

So, as I understand the `git` utility under Windows sometimes (or always) runs in a Linux-like shell; otherwise, the conversion to slashes would not work (Windows uses backslashes).

For the first replace (backslash-to-slash), Golang has a [filepath.ToSlash function](https://pkg.go.dev/path/filepath#ToSlash).

Then, a problem with spaces is IMHO not related to Windows, it seems generic enough to be handled outside this `if` clause.
I suspect this same problem could happen in Linux had the `git` utility contain a space in its path.
Thus, if I were you I would try putting the `helperName` into double quotes at lines 94 and 98 below.

Reply:
Tried that, git on Windows will then treat it as a string aka name, and will prepend it with git-credential again :joy:

Reply:
Did not know the ToSlash function. Will utilize that for the first part.

Reply:
Seems like the fix should be in the code that sets `helperPath` and not here. That way the user sees the correct default value in the usage message. Also - have you looked at `filepath.Abs`? I'd think that would work correctly on windows.

Reply:
I will look into `filepath.Abs` as well. To be clear, this format I need to put in there is specific for how `git` on Windows needs it to be in order to run. On a related note, we are [not the only ones struggling with spaces on Windows and commands and formatting](https://github.com/golang/go/issues/17149).

Reply:
`filepath.Abs` sadly does not give that specific format. :slightly_smiling_face:

Reply:
k. lgtm then. i'll let volodya approve

Reply:
Did you check if the Git on Linux processes spaces inside the path correctly?
I suspect it would have the same problem, in which case we need space escaping to be common.

Reply:
I never tried to do it in Linux yet, with absolute paths. So I will try and see what I can do there as well. I also realized I put the logic in the wrong place. The `helperName` should be changed, not `dst`. So I fixed that now. Tested it on Windows and it works correctly now. 

Let me see what happens on Linux when given a file path with spaces and absolute path.

Reply:
Added some logic that detects the usage of `creds-path`, then always use the full path. Most likely scenario on Linux is somebody does not have something in their `PATH` variable we can write to, like a fresh install of any distro and then they give the `--creds-path` and on that change we use the full path in the git config as well.

Reply:
I'm not sure I understand the `creds-path` logic; but I guess you know what you do here.

As long as it works I'm fine.